packages/eas-cli/src/commands/build/index.ts

```diff
@@ -95,6 +97,13 @@ export default class Build extends EasCommand {
 
   async runAsync(): Promise<void> {
     const { flags: rawFlags } = await this.parse(Build);
+
+    await maybeWarnAboutEasOutagesAsync(
```

Comment:
move this after L112. This has a the benefit of not hitting our API if the flags/project root dir are incorrect and the command will fail anyways.

Reply:
I agree that this improvement makes sense, although my only doubt is that in such a case this warning won't be the first thing that the user sees on the screen (because when `platform` flag is empty we ask the user about it). Example:

<img width="703" alt="Screenshot 2022-08-26 at 06 16 10" src="https://user-images.githubusercontent.com/55145344/186820931-6e72ca51-eded-4d0b-8da8-b7b8bab1215d.png">

I will consult it with @brentvatne

Reply:
ideally we can both run this **after** flags/project root dir validation and **before** we show any output or prompts. if that looks like it might take a bit too much work for this PR, then let's move it as @wschurman suggests and create a task on linear to follow up on that

Reply:
I think I was able to successfully refactor the code so we can do flags/project root dir validation first, then we show outage warning and finally we ask for inputs 🙂 Please, let me know what you think about this change.